site-src/geps/gep-1424.md

```diff
@@ -0,0 +1,384 @@
+# GEP-1426: HTTPRoute Mesh Binding
+
+* Issue: [#1294](https://github.com/kubernetes-sigs/gateway-api/issues/1294)
+* Status: Implementable
+
+## Overview
+
+Similar to how `HTTPRoutes` bind to `Gateways` and mange North/South traffic flows in Gateway API’s ingress use-case, it would be natural to adopt a similar model for traffic routing concerns in service mesh deployments. The purpose of this GEP is to add a mechanism to the Gateway API spec for the purpose of associating `HTTPRoutes` to a service mesh and offering a model for service owners to manage traffic splitting configurations.
+
+## Personas
+
+This GEP uses the [roles and personas](https://gateway-api.sigs.k8s.io/concepts/security-model/#roles-and-personas) defined in the Gateway API security model.
+
+## Goals
+
+* SHOULD allow `HTTPRoute` traffic rules to be configurable for a mesh service by the application owner/producer.
+* SHOULD allow control by the cluster operator (mesh administrator) to grant permission for whether `HTTPRoute` resources in a given namespace are allowed to configure mesh traffic routing.
+* MUST only be allowed to configure "producer" traffic rules for a `Service` in the same namespace as the `HTTPRoute`.
+    * Traffic routing configuration defined in this way SHOULD be respected by ALL consumer services in all namespaces in the mesh.
+* SHOULD only be allowed to direct traffic to `backendRefs` within the same namespace when used for configuration of mesh traffic.
```

Comment:
Fine with this short term but should be opened up in a fast followup IMO

Reply:
Opened https://github.com/kubernetes-sigs/gateway-api/issues/1479 to track this.

Reply:
```suggestion
* SHOULD only be allowed to direct traffic to `backendRefs` within the same namespace when used for configuration of mesh traffic, setting a `ResolvedRefs` condition with `status: "False"` and reason `RefNotPermitted` on the `HTTPRoute` if this is attempted.
```

Opened https://github.com/kubernetes-sigs/gateway-api/issues/1480 to capture this longer-term - I'm going to move this out of the Goals section and into a new section for constraints/restrictions if the suggested status condition here seems reasonable.